flink-runtime/src/main/java/org/apache/flink/runtime/security/contexts/HadoopSecurityContextFactory.java

```diff
@@ -53,17 +54,13 @@ public boolean isCompatibleWith(SecurityConfiguration securityConfig) {
 	}
 
 	@Override
-	public SecurityContext createContext(SecurityConfiguration securityConfig) {
+	public SecurityContext createContext(SecurityConfiguration securityConfig) throws SecurityContextInitializeException {
 		try {
-			Class.forName(
-				"org.apache.hadoop.security.UserGroupInformation",
-				false,
-				HadoopSecurityContextFactory.class.getClassLoader());
 			UserGroupInformation loginUser = UserGroupInformation.getLoginUser();
```

Comment:
I think the reflective lookup of the class `UserGroupInformation` is useless if we directly depend on `UserGroupInformation` here. If Hadoop is not on the classpath, then this whole class won't load due to this. I think this defies the purpose of the reflective lookup.

Reply:
This problem hasn't been introduced with this PR, though.

Reply:
yes this is indeed a problem from way back. +1 to address it in separate JIRA